src/main/java/smarthome/models/User.java

```diff
@@ -0,0 +1,91 @@
+package smarthome.models;
+
+/**
+ * Class of type User.
+ */
+public class User{
+    /**
+     * Creating a private int called id.
+     */
+    private int id;
+    /**
+     * Creating a private String called name.
+     */
+    private String name;
+    /**
+     * Creating a private string called role.
+     */
+    private String role;
+    /**
+     * Creating a Room called location.
+     */
+    private Room location;
+
+    /**
+     * Instantiates a new User.
+     *
+     * @param id       the int id
+     * @param name     the String name
+     * @param location the Room location
+     */
+    public User(int id, String name, Room location,String role) {
+        this.id = id;
+        this.name = name;
+        this.location = location;
+        this.role = role;
+    }
```

Comment:
Could you implement the id with an auto-increment feature using a private static counter so that we don't need an id argument in the constructor?
Also, I think it would be useful to have an overload of the constructor that doesn't require a location parameter so that we can create a user and set his location later.

Reply:
Probably easily doable with static variables.

Reply:
Made the changes you guys requested